src/api.js

```diff
@@ -844,6 +912,48 @@ Module["onRuntimeInitialized"] = function onRuntimeInitialized() {
         return stmt;
     };
 
+    /** Iterate over multiple SQL statements in a SQL string
+     @param {string} sql a string of SQL
+     @return {StatementIterator} the resulting statement iterator
+     */
+    Database.prototype["iterateStatements"] = function iterateStatements(sql) {
+        return new StatementIterator(sql, this);
+    };
+
+    /* Internal method to implement statement iteration */
+
+    Database.prototype["advanceIterator"] = function advanceIterator(iter) {
+        var stack = stackSave();
+        var pzTail = stackAlloc(4);
+        setValue(apiTemp, 0, "i32");
+        setValue(pzTail, 0, "i32");
+        var returnCode = sqlite3_prepare_v2(
+            this.db,
+            iter.nextSql,
+            -1,
+            apiTemp,
+            pzTail
+        );
+        var tail = UTF8ToString(getValue(pzTail, "i32"));
+        iter.lastSql = iter.nextSql.substr(
+            0,
+            iter.nextSql.length - tail.length
+        );
+        iter.nextSql = tail;
+        stackRestore(stack);
+        if (returnCode === SQLITE_OK) {
```

Comment:
Can't we use the existing error handling method ?

Reply:
Yes/no.  I have to check the return code manually anyway, because if there is an error, I still need to grab the last SQL command to put in the iterator object.  If I simply hand off to handleError(), then there is no opportunity to do that - the user will get an error message, and have no way of finding out the SQL text that actually generated the error (which is a problem for my use case). I can call handleError() after that, but it seemed kind of pointless.

Reply:
I do not understand. `handleError(returnCode); happy_path()` would do the exact same thing as `if (returnCode === ok) { happy_path() } manual_error_handling() `, it would just be clearer.

Reply:
Oops, you're absolutely right.  I got confused in my own code.